src/hex/coordinate.rs

```diff
@@ -202,6 +354,148 @@ impl Axial {
         self.make_vector(1, direction.into())
     }
 
+    /// Get all the neighbors for this coordinate.
+    ///
+    /// See [`HexDirection`] for a reference of directionality.
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial};
+    ///
+    /// let coord = axial!(1, 2);
+    ///
+    /// if axial!(0, 0).neighbors().contains(&coord) {
+    ///     // Do something ...
+    /// }
+    /// ```
+    pub fn neighbors(&self) -> [Self; 6] {
+        [
+            self.neighbor(HexDirection::Front),
+            self.neighbor(HexDirection::FrontRight),
+            self.neighbor(HexDirection::BackRight),
+            self.neighbor(HexDirection::Back),
+            self.neighbor(HexDirection::BackLeft),
+            self.neighbor(HexDirection::FrontLeft),
+        ]
+    }
+
+    /// Checks if ALL of the supplied coordinates are neighbors to self.
+    ///
+    /// See [`Axial::neighbor`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial};
+    ///
+    /// let coord = axial!(1, 2);
+    ///
+    /// if axial!(0, 0).are_neighbors(&[coord]) {
+    ///     // Do something ...
+    /// }
+    /// ```
+    pub fn are_neighbors(&self, coords: &[Self]) -> bool {
+        for coord in coords {
+            if !self.neighbors().contains(coord) {
+                return false;
+            }
+        }
+        true
+    }
+
+    /// Generates all 6 vertices that are associated with this tile.
+    ///
+    /// See [`Vertex`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial, Vertex};
+    ///
+    /// axial!(0, 0).vertices().iter().map(|vert| {/* ... */} );
+    ///
+    /// ```
+    pub fn vertices(&self) -> [Vertex; 6] {
+        [
+            vertex!(self.q, self.r, VertexSpin::Up),
+            vertex!(self.q + 1, self.r - 1, VertexSpin::Down),
+            vertex!(self.q, self.r + 1, VertexSpin::Up),
+            vertex!(self.q, self.r, VertexSpin::Down),
+            vertex!(self.q - 1, self.r + 1, VertexSpin::Up),
+            vertex!(self.q, self.r - 1, VertexSpin::Down),
+        ]
+    }
+
+    /// Given two neighboring tiles produce the shared vertices.
+    ///
+    /// See [`Vertex`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial, Vertex};
+    ///
+    /// let verts = axial!(0, 0).shared_vert_two(axial!(1, 0));
+    ///
+    /// ```
+    pub fn shared_vert_two(&self, b: Self) -> Option<[Vertex; 2]> {
```

Comment:
Confirming for my own understanding: are these the same position in space, but represented in relation to two different hexagons?